### ISSUE TITLE
Find a way of managing state in .rsr files.

### ISSUE THREAD

**Issue by ContainedBlargh** [ContainedBlargh/dirthouse#1]
In `actix-web`, state is managed by declaring a `actix_web::web::Data<T>` argument in your Responder implementation functions. That data needs to be initialized in the main method and registered with the actix App object, but otherwise, it should be fairly simple to pass data around in actix.

However, there is no way to do this in DIRTHOUSE (dh) so far.
It would be great if we could have a central place where such a state is defined, preferably in a regular .rs file such that it can be passed to any handler that may need to use it. It could e.g. be maintaining a database connection or something similar.

Alternatively, each `.rsr`-file/module could expose a state object that should be maintained somewhere else.

In my example I have created persistent state using the lazy_static package, but that's a horrible hack and not suitable in the long run.

In short, it would be nice to come up with a plan for managing state.
